### PR TITLE
[stark] Fourier PE frequency sweep: n=6 vs n=16 (baseline n=8) (Wave 21)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The current best model uses `FourierEmbed` with **8 frequency bands** (default). This encodes 3D coordinates as `[sin(2^k * π * x), cos(2^k * π * x)]` for k=0..7, giving 48 Fourier features projected to 256d (12,544 params in the input projection).

The number of frequency bands is an unexplored hyperparameter. Too few frequencies → poor spatial resolution, especially for high-frequency boundary features (shear layers, separation zones). Too many → overfit to noise, harder to train. The wall-shear axes (wsy, wsz) are dominated by high-frequency spatial structure near walls and edges — more frequency bands may capture these better.

We sweep **3 values**: `--fourier-pe-num-freqs 6` (fewer), `--fourier-pe-num-freqs 8` (baseline), `--fourier-pe-num-freqs 16` (more). Run arms A (n=6) and B (n=16) — arm baseline (n=8) is already known from PR #174.

## Instructions

Run **two training arms** using the canonical 5L/256d baseline config, changing only `--fourier-pe-num-freqs`:

**Arm A — fewer freqs (n=6):**
```bash
torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 --model-hidden-dim 256 --model-heads 4 \
  --fourier-pe --fourier-pe-num-freqs 6 \
  --no-use-ema \
  --lr 3e-4 --lr-cosine-t-max 50 \
  --no-compile-model \
  --wandb-group bengio-wave21-stark-fourier-freq
```

**Arm B — more freqs (n=16):**
```bash
torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 --model-hidden-dim 256 --model-heads 4 \
  --fourier-pe --fourier-pe-num-freqs 16 \
  --no-use-ema \
  --lr 3e-4 --lr-cosine-t-max 50 \
  --no-compile-model \
  --wandb-group bengio-wave21-stark-fourier-freq
```

Apply standard kill gates to both arms independently. If Arm B (n=16) is clearly dominant at EP10, you may optionally drop Arm A and run both GPUs on a single arm.

Report per-axis breakdown (wsy, wsz) in results — the hypothesis is specifically that more frequencies help the high-frequency wsy/wsz axes.

Note: n=16 will increase input_projection size: 3 features × 16 freqs × 2 = 96 → Linear(96, 256) = 24,832 params vs 12,544 for n=8. Total model grows by ~12k params (~0.3%) — negligible architecture change.

## Kill gates (abupt, applied per arm)
- EP5: kill if > 13%
- EP10: kill if > 10.5%
- EP15: kill if > 8.5%
- EP20: kill if > 8.0%
- EP25: kill if ≥ 6.9549% (current baseline)

## Baseline

Current best: **PR #174** (alphonse, 5L/256d/4H + FourierEmbed n=8 + T_max=50, no-EMA)
W&B run: `vu4jsiic`, ep~45.3, step 807,025

| Metric | Current Best (val) | AB-UPT Target |
|--------|-------------------|---------------|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **6.9549%** | 4.51% |
| `val_primary/surface_pressure_rel_l2_pct` | 4.5644% | 3.82% |
| `val_primary/volume_pressure_rel_l2_pct` | **3.9361%** ✓ | 6.08% |
| `val_primary/wall_shear_y_rel_l2_pct` | 8.7345% | 3.65% |
| `val_primary/wall_shear_z_rel_l2_pct` | 10.5766% | 3.63% |

Binding constraint: wsy (5.08pp gap) and wsz (6.95pp gap).

Reproduce command (baseline n=8):
```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 --model-hidden-dim 256 --model-heads 4 \
  --fourier-pe --no-use-ema \
  --lr 3e-4 --lr-cosine-t-max 50 \
  --no-compile-model
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT stark: The `senpai-bengio-stark` deployment is not yet provisioned — infrastructure issue #466 tracks this. You cannot launch until the pod is created. No action required on your end; this is blocked on the human infrastructure team.

Once your pod is provisioned, launch immediately with the Fourier PE sweep from the PR body. Notify in this PR when you start.
